### PR TITLE
add rocm GHA workflow

### PR DESCRIPTION
**Pull request by jeffdaily**
cc @jeffdaily @sunway513 @jithunnair-amd @ROCmSupport @KyleCZH

### CODE REVIEW

File: .github/scripts/generate_ci_workflows.py [hunk 1]
```diff
@@ -576,6 +595,16 @@ def generate_workflow_file(self, workflow_template: jinja2.Template) -> None:
...
         ),
     ),
+    CIWorkflow(
+        arch="linux",
+        build_environment="linux-bionic-rocm4.5-py3.7",
+        docker_image_base=f"{DOCKER_REGISTRY}/pytorch/pytorch-linux-bionic-rocm4.5-py3.7",
+        test_runner_type=LINUX_ROCM_TEST_RUNNER,
+        num_test_shards=2,
+        ciflow_config=CIFlowConfig(
+            labels=set([LABEL_CIFLOW_DEFAULT, LABEL_CIFLOW_LINUX, LABEL_CIFLOW_CUDA]),
```
Comment (seemethere) [inline .github/scripts/generate_ci_workflows.py:605]:
```suggestion
            labels=set([LABEL_CIFLOW_LINUX, LABEL_CIFLOW_ROCM]),
```

Let's add another CIFLOW label above and then limit these only to run when `LABEL_CIFLOW_ROCM` is actually active

**Review by seemethere — COMMENTED** [review]
From our sync up:

* Let's change the default target for RoCM workflows from `pull_request` -> `pull_request_target`
* Move to only trigger on PRs with `ciflow/rocm` label

Once we can confirm that builds do indeed get secrets on `pull_request_target` triggers we can then enable them and start to move the bulk of the rocm infra over

File: .github/scripts/generate_ci_workflows.py (hunk 1 above)
Comment (jeffdaily) [reply .github/scripts/generate_ci_workflows.py]: 59dbc9c2108331dd02ee845371c67a58f61d962c creates the new LABEL_CIFLOW_ROCM.

**Comment by jeffdaily** [comment]
@seemethere can confirm ciflow/rocm label is working.  I did not change from pull_request to pull_request_target; I think pull_request should still work for forked repos, or at least I'd like to try with a follow-on test PR.  Please merge if you approve the rest of the changes.

**Comment by janeyx99** [comment]
@jeffdaily So the pull_request trigger doesn't get access to secrets whereas the pull_request_target trigger does. Are there so secrets that the ROCm workflows need to access?

**Comment by jeffdaily** [comment]
@janeyx99 here is a link to the latest version of the rocm workflow file from this PR:

https://github.com/pytorch/pytorch/blob/0d066aa1a40e50c4629905325a1c9d53a31139e5/.github/workflows/generated-linux-bionic-rocm4.5-py3.7.yml

The secrets I see are GITHUB_TOKEN and SCRIBE_GRAPHQL_ACCESS_TOKEN.  Chances are the latter secret would fail for a forked pytorch repo, but the GITHUB_TOKEN will exist for forked repos and should provide at least read-only access.

I hope I'm not being overly paranoid, but my concern with using pull_request_trigger is that well all might not fully understand the security implications.  I don't want to inadvertently cause harm by using it.

File: .github/templates/common.yml.j2 [hunk 2]
```diff
@@ -103,6 +103,46 @@ concurrency:
           env | grep '^GITHUB' > "/tmp/github_env_${GITHUB_RUN_ID}"
 {%- endmacro -%}
 
+<<<<<<< HEAD
```
Comment (seemethere) [inline .github/templates/common.yml.j2:106]: Oh looks like this may be a leftover

**Review by seemethere — COMMENTED** [review]
Looks mostly good just one leftover merge conflict thing and I think we can get this in

**Review by seemethere — APPROVED** [review]

File: .github/templates/common.yml.j2 (hunk 2 above)
Comment (jeffdaily) [reply .github/templates/common.yml.j2]: thanks for fixing that.

**Pull request closed without merging**